Signal/src/util/DateUtil.m

```diff
@@ -25,6 +25,64 @@ + (NSDateFormatter *)dateFormatter {
     return formatter;
 }
 
++ (NSDateFormatter *)dateBreakRelativeDateFormatter
+{
+    static NSDateFormatter *formatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        formatter = [NSDateFormatter new];
+        formatter.locale = [NSLocale currentLocale];
+        formatter.dateStyle = NSDateFormatterShortStyle;
+        formatter.timeStyle = NSDateFormatterNoStyle;
+        formatter.doesRelativeDateFormatting = YES;
+    });
+
+    return formatter;
+}
+
++ (NSDateFormatter *)dateBreakThisWeekDateFormatter
+{
+    static NSDateFormatter *formatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        formatter = [NSDateFormatter new];
+        formatter.locale = [NSLocale currentLocale];
+        // "Monday", "Tuesday", etc.
+        formatter.dateFormat = @"EEEE";
+    });
+
+    return formatter;
+}
+
++ (NSDateFormatter *)dateBreakThisYearDateFormatter
+{
+    static NSDateFormatter *formatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        formatter = [NSDateFormatter new];
+        formatter.locale = [NSLocale currentLocale];
+        // Tue, Jun 6
+        formatter.dateFormat = @"EE, MMM d";
```

Comment:
I wonder if we should use this kind of complicated date format - it won't necessarily localize well.

Reply:
Yeah, same issue with the message footer stuff we merged yesterday. Suggestion?

Reply:
Yup - that's why I did a hybrid thing there and tried to used a localize format as much as possible.  No change requested, but we should mention this to Myles.